### ISSUE TITLE
error: Missing type parameters for generic type "array"

### ISSUE THREAD

**Issue by sg495** [python/mypy#13942]
**Bug Report**

Mypy strict mode complains if a type parameter is omitted  when using the built-in `array` type in an annotation, while Python raises `TypeError` if a type parameter is included. Mypy only complains in strict mode.

**To Reproduce**

```python
from array import array
x: array = array("L")      # Mypy error: Missing type parameters for generic type "array"
y: array[int] = array("L") # TypeError: 'type' object is not subscriptable
```

**Expected Behavior**

Mypy strict mode should not complain about a missing generic type parameter (since Python does not allow it).

**Actual Behavior**

```
array-mypy.py:2: error: Missing type parameters for generic type "array"
```

**Possible Workaround**

```python
from array import array
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    IntArray = array[int]
else:
    IntArray = array

x: IntArray = array("L")

```

**Your Environment**

- Mypy version used: `mypy 0.982 (compiled: yes)`
- Mypy command-line flags: `mypy --strict`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: `3.9.13`

**Comment by JelleZijlstra** [comment]
See https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime for general guidance.

However, in this case we should probably make `array` support `__class_getitem__` at runtime. I can't find a reason why we didn't do that before. I'll submit an issue to CPython to make that change, but it won't be released until 3.12.
